diff --git a/Makefile b/Makefile
index fabb381..a29dbba 100644
--- a/Makefile
+++ b/Makefile
@@ -11,8 +11,11 @@ PREFIX  = arm-vita-eabi
 CC      = $(PREFIX)-gcc
 CXX     = $(PREFIX)-g++
 AR      = $(PREFIX)-gcc-ar
+VGL_GIT_HASH ?= $(shell git rev-parse --short HEAD 2>/dev/null)
+EXTRA_CFLAGS ?=
 CFLAGS  = -g -Wl,-q -O3 -ffast-math -mtune=cortex-a9 -mfpu=neon -Wno-incompatible-pointer-types -Wno-stringop-overflow -mfp16-format=ieee \
-	-DVGL_GIT_HASH='"$(shell git rev-parse --short HEAD)"' -Isource
+	-DVGL_GIT_HASH='"$(VGL_GIT_HASH)"' -Isource
+CFLAGS += $(EXTRA_CFLAGS)
 ASFLAGS = $(CFLAGS)
 
 ifeq ($(SOFTFP_ABI),1)
@@ -275,7 +278,7 @@ CXXFLAGS  = $(CFLAGS) -fexceptions -std=gnu++11 -Wno-write-strings
 all: $(TARGET).a
 
 $(TARGET).a: $(OBJS)
-	$(AR) -rc $@ $^
+	$(AR) -rcD $@ $^
 	
 %.smpc:
 	@make -C $(@:.smpc=) clean
diff --git a/source/vgl.c b/source/vgl.c
index 5c4f15a..608df61 100644
--- a/source/vgl.c
+++ b/source/vgl.c
@@ -303,7 +303,7 @@ GLboolean vglInitWithCustomSizes(int pool_size, int width, int height, int ram_po
 	blit_attrs[0].regIndex = sceGxmProgramParameterGetResourceIndex(blit_position);
 	blit_attrs[1].regIndex = sceGxmProgramParameterGetResourceIndex(blit_texcoord);
 	blit_attrs[0].streamIndex = blit_attrs[1].streamIndex = 0;
-	blit_streams[0].indexSource = blit_streams[1].indexSource = SCE_GXM_INDEX_SOURCE_INDEX_16BIT;
+	blit_streams[0].indexSource = SCE_GXM_INDEX_SOURCE_INDEX_16BIT;
 	{ patch_vertex_program(gxm_shader_patcher, blit_vertex_id, blit_attrs, 2, blit_streams, 1, &blit_vertex_program_patched); }
 	{ patch_fragment_program(gxm_shader_patcher, blit_fragment_id, SCE_GXM_OUTPUT_REGISTER_FORMAT_UCHAR4, msaa_mode, NULL, NULL, &blit_fragment_program_patched); }
 	{ patch_fragment_program(gxm_shader_patcher, blit_fragment_id, SCE_GXM_OUTPUT_REGISTER_FORMAT_HALF4, msaa_mode, NULL, NULL, &blit_fragment_program_float_patched); }
